src/qml/BusyOverlay.qml

```diff
@@ -69,13 +69,34 @@ Rectangle {
     to: 1.0
   }
 
-  Text {
-    id: busyMessage
+  Rectangle{
+    id: busyMessageShield
     anchors.top: busyIndicator.bottom
+    anchors.topMargin: 10
     anchors.horizontalCenter: parent.horizontalCenter
-    horizontalAlignment: Text.AlignHCenter
-    font: Theme.tipFont
-    color: Theme.mainColor
-    text: ''
+    color: Theme.darkGraySemiOpaque
+    radius: 4
+
+    width: busyMessage.width
+    height: busyMessage.contentHeight + 5
+
+    Text {
+      id: busyMessage
+
+      property int absoluteWidth: busyMessageFontMetrics.boundingRect(text).width + 10
+      width: Math.min(mainWindow.width - 20, absoluteWidth)
+
+      anchors.centerIn: parent
+      horizontalAlignment: Text.AlignHCenter
+      font: Theme.tipFont
+      color: "white"
```

Comment:
I guess hardworking this as white will always work fine with  `Theme.darkGraySemiOpaque`. Any chance we can change this hard-coded color to something from theme?

Reply:
@suricactus , I'd like to defer that to a/ a different PR, and b/ whenever we'll need to change throughout the code. We use "white" for all tool buttons too, which is the main context this color is used against. 

Point taken though :)